cli/command/container/utils.go

```diff
@@ -38,6 +49,17 @@ func waitExitOrRemoved(ctx context.Context, dockerCli *command.DockerCli, contai
 			if !waitRemove {
 				stopProcessing = true
 			}
+
+			// If we are talking to an older daemon, `AutoRemove` is not supported.
+			// We need to fall back to the old behavior, which is client-side removal
+			if waitRemove && versions.LessThan(dockerCli.Client().ClientVersion(), "1.25") {
```

Comment:
I think this should be `dockerCli.Client().ServerVersion().Version`, it should detect daemon version but not the client version.